src/components/AddBook/AddBook.jsx

```diff
@@ -0,0 +1,108 @@
+import { useState, useEffect } from "react";
+import "./AddBook.css"
+
+function AddBook(){
+
+    const [title, setTitle] = useState('')
+    const [author, setAuthor] = useState('')
+    const [genre, setGenre] = useState('')
+
+    // Genre handling
+    const [genresList, setGenresList] = useState([])
+    const [genresListLength, setGenresListLength] = useState(0)
+
+    // Optional
+    const [blurb, setBlurb] = useState('')
+        const [isBlurb, setIsBlurb] = useState(false)
+    const [imageurl, setImageurl] = useState('')
+        const [isImageUrl, setIsImageUrl] = useState(false)
+    const [year, setYear] = useState('')
+        const [isYear, setIsYear] = useState(false)
+    
+    
+    // Fetch to get genres:
+    useEffect(() => {
+        fetch("https://book-swap-api.dev.io-academy.uk/api/genres")
+            .then((response) => {
+                return response.json()
+            })
+                .then((genres_json) => {
+                    setGenresList(genres_json.data)
+                    setGenresListLength(genres_json.data.length)  
+                })
+    }, [])
+
+
+    function addBookSubmit(event){
+        event.preventDefault()
+        let bookData = {
+            'title': title,
+            'author': author,
+            'genre_id': parseInt(genre)
+        }
+
+        if (blurb != '') {
+            bookData['blurb'] = blurb
+        }
+        if (imageurl != '') {
+            bookData['image'] = imageurl
+        }
+        if (year != '') {
+            bookData['year'] = year
+        }
+        
+        console.log(`title: ${title}, author: ${author}, genre_id: ${genre}, blurb: ${blurb}, image: ${imageurl}, year: ${year}`)
+        fetch ("https://book-swap-api.dev.io-academy.uk/api/books", {
+            method: 'POST',
+            mode: 'cors',
+            headers: {'content-type':'application/json', 'accept':'application/json'
+        },
+            body: JSON.stringify(bookData)
+        })
+            
+            .then(function (res) {
+                if (res.status === 201) {
+                    // Book added successfully
+                    console.log('Book added successfully.');
+                } else if (res.status === 500 || res.status === 422) {
+                    // Handle errors (server error or validation errors) appropriately
+                    console.error('Error adding the book.');
+                } else {
+                    // Handle other status codes as needed
+                    console.error('Unexpected error.');
```

Comment:
Error handling is decent, but now you need to actually display these messages somewhere